### ISSUE TITLE
Local build doesn't work, but homebrew version works fine

### ISSUE THREAD

**Issue by subodh-dharma** [awslabs/amazon-ecr-credential-helper#159]
I tried cloning the repo and perform a local build with `make docker`. Placed the docker-credential-ecr-login binary in the PATH, and after all the configs suggested in the README, it didn't work. It continues to give the error.

```
Error response from daemon: Get https://amazon-id.dkr.ecr.us-west-2.amazonaws.com/v2/nginx/manifests/1.12.2: no basic auth credentials
```

I downloaded the homebrew version, it works perfectly fine. The homebrew version binary is 0.2.0 while that when I build locally is 0.3.0

**Comment by samuelkarp** [comment]
@subodh-dharma 	I don't have a Mac, so my ability to help you debug is going to be a bit more limited.  Can you check the logs in `~/.ecr`?  There should be more information on what is failing there.

**Comment by subodh-dharma** [comment]
@samuelkarp The `~/.ecr` directory didn't have any log files, just the cache.json from the helper installed from the homebrew.

**Comment by ktax** [comment]
Hey! 
I had the same problem and I needed to set up aws credentials in the ~/.aws/credentials instead of ENV Variables. When installed with homebrew it works for both settings. Hope it helps you!

**Comment by samuelkarp** [comment]
@ktax @subodh-dharma Building from source should work as well, so I don't currently know what's wrong.

> @samuelkarp The ~/.ecr directory didn't have any log files, just the cache.json from the helper installed from the homebrew.

That's odd.  I've noticed that sometimes log lines are output to stderr (and to the terminal) instead of the file, which looks like an error either in how I've configured the logs or in the log library; I'll dig into this more (though unfortunately I don't have a timeline for when I'll be able to look).

**Comment by ktax** [comment]
@samuelkarp I've just removed my ~/aws/credentials file and moved my credentials back to env variables to be sure. As I said it does not work for helper built from source (at least for me). It works for helper installed with homebrew though.  I think it is worth a try!

**Comment by subodh-dharma** [comment]
@samuelkarp -- what is the OS you are working on. I can try spinning up a VM and try building on that and see if the issue still exists.

**Comment by samuelkarp** [comment]
@subodh-dharma My personal development environment tends to be Ubuntu 16.04 with Go installed into my home directory and Docker installed from the [Docker `apt` repo](https://docs.docker.com/install/linux/docker-ce/ubuntu/#install-docker-ce-1).

**Comment by samuelkarp** [comment]
@subodh-dharma The issue with the log files has been fixed with the most recent version, [v0.3.1](https://github.com/awslabs/amazon-ecr-credential-helper/releases/tag/v0.3.1).  Can you try this version and see if you're still experiencing problems?

**Comment by samuelkarp** [comment]
We haven't heard back from you in a while, so I'm closing this issue.  If you continue to have trouble, please open a new issue.
